src/ParaTest/Runners/PHPUnit/ResultPrinter.php

```diff
@@ -201,16 +201,6 @@ public function printFeedback(ExecutableTest $test)
                 $test->getPath()
             ));
         }
-        if (!$reader->hasResults()) {
```

Comment:
Thanks for the feedback. I looked into it. The reader still detects process crashes. It checks if the file size is 0 in ParaTest\Logging\JUnit\Reader on line 46, whereas hasResult() checks whether there is empty XML (but valid) in the file. If it is invalid XML, instantiating the SimpleXmlElement object would fail (which is not caught by the way, so there is probably no use case for that).

An empty XML file indicates there was no test executed. An entirely empty file or invalid XML lets one assume the process crashed or went mad or something.

I do not really understand what you mean with "more than one class in the file". Do mean several test cases in one unit test file? I tried both, having several unit tests in one test file, which results in only the first being executed and I tried to place a second class in the file of the class to test. This did not effect results at all.

At least, this is what my quick tests with phpunit 5.7 yielded. It's likely I oversaw something.

Could you please point me to such edge cases?

It is really a pity, as we introduced test groups in our project to be able to run specific tests only to reduce overall execution time. But this is of no use, as we cannot use paratests anymore and have to rely on phpunit only instead. Less tests are executed, but they are not run simultaneously anymore. You probably get the issue here :-)